src/Generator/Generators/CSharp/CSharpTypePrinter.cs

```diff
@@ -686,6 +686,31 @@ private static string GetParameterUsage(ParameterUsage usage)
         }
 
         private CSharpExpressionPrinter expressionPrinter => new CSharpExpressionPrinter(this);
+
+        public override TypePrinterResult VisitFieldDecl(Field field)
+        {
+            List<TranslationUnit> unit = new List<TranslationUnit>();
+            var cs = new CSharpSources(Context, unit);
+            var safeIdentifier = cs.SafeIdentifier(field.Name);
+
+            PushMarshalKind(MarshalKind.NativeField);
+            var fieldTypePrinted = field.QualifiedType.CSharpType(this);
+            PopMarshalKind();
+
+            var fieldType = field.QualifiedType.Type.Desugar().IsAddress() ?
+                CSharpTypePrinter.IntPtrType : fieldTypePrinted.Type;
+
+            var fieldName = safeIdentifier;
+            if (!string.IsNullOrWhiteSpace(fieldTypePrinted.NameSuffix))
+                fieldName += fieldTypePrinted.NameSuffix;
```

Comment:
You add the suffix twice, once here and once in returnTypePrinter.NameSuffix. Select just one.